drf_kit/managers/availability_managers.py

```diff
@@ -0,0 +1,117 @@
+from dataclasses import dataclass
+from datetime import datetime
+
+from django.db import models
+from django.db.models import Q
+from django.utils import timezone
+
+
+@dataclass
+class AvailabilityChecker:
+    starts_at: datetime | None = None
+    ends_at: datetime | None = None
+    at: datetime | None = None
+
+    def __post_init__(self):
+        self.at = self.at or timezone.now()
+
+    @property
+    def is_future(self) -> bool:
+        not_started = self.future_start
+        not_ended = self.future_end or self.undefined_end
+        return not_started and not_ended
+
+    @property
+    def is_past(self) -> bool:
+        started = self.past_start or self.undefined_start
+        ended = self.past_end
+        return started and ended
+
+    @property
+    def is_current(self) -> bool:
+        started = self.past_start or self.undefined_start
+        not_ended = self.future_end or self.undefined_end
+        return started & not_ended
+
+    @property
+    def undefined_start(self) -> bool:
+        return self.starts_at is None
+
+    @property
+    def undefined_end(self) -> bool:
+        return self.ends_at is None
+
+    @property
+    def past_start(self) -> bool:
+        return not self.undefined_start and self.starts_at < self.at
+
+    @property
+    def past_end(self) -> bool:
+        return not self.undefined_end and self.ends_at < self.at
+
+    @property
+    def future_start(self) -> bool:
```

Comment:
What about the `starts_at` is equal to `at`?

Reply:
Thanks, I was not covering an exact match. Which is very rare considering datetime and now (with millisecond precision)

I added a new commit just to handle that.